### PR TITLE
8340477: Remove JDK1.1 compatible behavior for "EST", "MST", and "HST" time zones

### PR DESCRIPTION
**Pull request by naotoj**
Removing the old compatibility behavior switch for JDK1.1. With the integration of tzdata2024b, definitions of those three (deprecated) IDs changed again, so maintaining the JDK11 compatibility got less useful. Removing the compatibility behavior makes the code and test simpler. A CSR for the behavior change is drafted too.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8340478](https://bugs.openjdk.org/browse/JDK-8340478) to be approved

### Issues
 * [JDK-8340477](https://bugs.openjdk.org/browse/JDK-8340477): Remove JDK1.1 compatible behavior for "EST", "MST", and "HST" time zones (**Bug** - P4)
 * [JDK-8340478](https://bugs.openjdk.org/browse/JDK-8340478): Remove JDK1.1 compatible behavior for "EST", "MST", and "HST" time zones (**CSR**)

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21545/head:pull/21545` \
`$ git checkout pull/21545`

Update a local copy of the PR: \
`$ git checkout pull/21545` \
`$ git pull https://git.openjdk.org/jdk.git pull/21545/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21545`

View PR using the GUI difftool: \
`$ git pr show -t 21545`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21545.diff">https://git.openjdk.org/jdk/pull/21545.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21545#issuecomment-2417534375)

### CODE REVIEW

File: src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java [hunk 1]
```diff
@@ -275,21 +232,13 @@ public Void run() {
...
-            aliases.put("EST", "America/New_York");
-            aliases.put("MST", "America/Denver");
-        } else {
-            aliases.put("EST", "America/Panama");
-            aliases.put("MST", "America/Phoenix");
+        for (var key : ZoneId.SHORT_IDS.keySet()) {
+            switch (key) {
+                case "EST" -> aliases.put("EST", "America/Panama");
+                case "MST" -> aliases.put("MST", "America/Phoenix");
+                default -> aliases.put(key, ZoneId.SHORT_IDS.get(key));
```
Comment (justin-curtis-lu) [inline src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java:239]: "EST" and "MST" case are no longer needed? ZoneId should already map them to "America/Panama" and "America/Phoenix". We no longer offer the old mapping, so we can safely follow ZoneId.SHORT_IDS.

Comment (naotoj) [reply src/java.base/share/classes/sun/util/calendar/ZoneInfoFile.java]: Right. Removed them.

**Review by irisclark — APPROVED** [review]
Happy to see this 1.1-era behaviour support removed.

**Review by justin-curtis-lu — APPROVED** [review]
LGTM

**Comment by justin-curtis-lu** [comment]
BTW, can we go even further and remove "EST", "HST" and "MST" from `SHORT_IDS`? IIUC the basis for keeping those 3 specifically as part of `SHORT_IDS` was the conflicting 1.1 behavior. Now that the old mapping is removed, those 3 will always map to the TZDB data, which has them as valid links in _backwards_ instead of the previous fixed offset zones.

Or is there some other reason to keep them at this point? I guess there is also no benefit to incentivize the usage of _any_ 3 letter zone ID.

**Comment by naotoj** [comment]
> BTW, can we go even further and remove "EST", "HST" and "MST" from `SHORT_IDS`? IIUC the basis for keeping those 3 specifically as part of `SHORT_IDS` was the conflicting 1.1 behavior. Now that the old mapping is removed, those 3 will always map to the TZDB data, which has them as valid links in _backwards_ instead of the previous fixed offset zones.

Data conflict with TZDB is not the reason to keep them in `SHORT_IDS`, but they are defined as time zone ids in JDK1.1, as well as others in `SHORT_IDS` is the reason. They are all deprecated, so ideally we remove them altogether, but I expect that would break apps.

That said, it might be useful we log warnings if apps use them, then we may be able to indeed remove them in future releases.

**Comment by naotoj** [comment]
Issue created for those IDs to be removed: https://bugs.openjdk.org/browse/JDK-8342550

**Review by JoeWang-Java — APPROVED** [review]
Nice cleanup.

**Comment by naotoj** [comment]
Thanks for the reviews!
/integrate

**Pull request closed without merging**